### PR TITLE
Fix compatibility issue with Openfire 4.9.0

### LINKED ISSUE #387: Compatibility issue with Openfire 4.9.0
Version 2.5.0 fails to load in Openfire 4.9.0. This error is logged:

```
org.jivesoftware.openfire.container.PluginManager - An exception occurred while loading plugin 'monitoring':
java.lang.NoSuchMethodError: 'java.lang.String org.jivesoftware.util.JiveGlobals.getHomeDirectory()'
        at org.jivesoftware.openfire.archive.ArchiveIndexer.<init>(ArchiveIndexer.java:70) ~[?:?]
        at org.jivesoftware.openfire.plugin.MonitoringPlugin.<init>(MonitoringPlugin.java:128) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:?]
        at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at org.jivesoftware.openfire.container.PluginManager.loadPlugin(PluginManager.java:587) [xmppserver-4.9.0-SNAPSHOT.jar:4.9.0-SNAPSHOT]
        at org.jivesoftware.openfire.container.PluginMonitor$MonitorTask$4.call(PluginMonitor.java:380) [xmppserver-4.9.0-SNAPSHOT.jar:4.9.0-SNAPSHOT]
        at org.jivesoftware.openfire.container.PluginMonitor$MonitorTask$4.call(PluginMonitor.java:368) [xmppserver-4.9.0-SNAPSHOT.jar:4.9.0-SNAPSHOT]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

### PR DESCRIPTION
**Pull request by guusdk**
Replaced Openfire API usage that was deprecated in Openfire 4.8.0 and removed in 4.9.0.

This plugin is now compatible with Openfire 4.9.0 and requires 4.8.0 or later. No longer compatible with versions older than 4.8.0.

fixes #387

### CODE REVIEW

**Comment by akrherz** [comment]
I think the [ssl repo issue for jrobin](https://opennms.discourse.group/t/ssl-certificate-issue-with-repo-opennms-org/3669) is fixed now, at least, but that does not seem to be the issue here, but will try pushing a change

**Comment by akrherz** [comment]
The OpenNMS repo appears empty / defunct.  I found a forked jrobin on github, so why not...

File: src/java/org/jivesoftware/openfire/plugin/MonitoringPlugin.java [hunk 1]
```diff
@@ -170,28 +173,22 @@ public void initializePlugin(PluginManager manager, File pluginDirectory) {
...
-        File pluginDir = new File(JiveGlobals.getHomeDirectory(), "plugins");
-        File[] jars = pluginDir.listFiles(new FileFilter() {
-            public boolean accept(File pathname) {
-                String fileName = pathname.getName().toLowerCase();
-                return (fileName.equalsIgnoreCase("enterprise.jar"));
-            }
-        });
-        if (jars.length > 0) {
+        // Check if we Enterprise is installed and stop loading this plugin if found
+        if (manager.getPluginByName("enterprise").isPresent()) {
```
Comment (Fishbowler) [inline src/java/org/jivesoftware/openfire/plugin/MonitoringPlugin.java:177]: Is this still necessary?
Would this plugin be compatible with 4.8?

**Review by Fishbowler — DISMISSED** [review]
Looks good. One question, but possibly scope creep either way :)

File: src/java/org/jivesoftware/openfire/plugin/MonitoringPlugin.java (hunk 1 above)
Comment (guusdk) [reply src/java/org/jivesoftware/openfire/plugin/MonitoringPlugin.java]: I've removed this bit. Good catch.

**Review by akrherz — APPROVED** [review]

**Pull request merged**